packages/story-editor/src/components/fontCheck/index.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useEffect, useCallback, useState } from '@googleforcreators/react';
+
+/**
+ * Internal dependencies
+ */
+import { useFeature } from 'flagged';
+import { getInUseFontsForPages } from '../../utils/getInUseFonts';
+import { useStory, useAPI } from '../../app';
+import { FontCheckDialog } from './fontCheckDialog';
+
+export const FontCheck = () => {
+  const isNotifyDeletedEnabled = useFeature('notifyDeletedFonts');
+  const storyPages = useStory(({ state: { pages } }) => pages);
+  const {
+    actions: { getFonts },
+  } = useAPI();
+
+  const [showDialog, setShowDialog] = useState({
+    visible: false,
+    missingFont: '',
+  });
+
+  const closeDialog = useCallback(
+    () => setShowDialog({ visible: false, missingFont: '' }),
+    []
+  );
+
+  useEffect(() => {
+    const checkFonts = async () => {
+      const inUseFonts = getInUseFontsForPages(storyPages);
+      const allFonts = await getFonts({ include: inUseFonts.join(',') });
+
+      const diff = inUseFonts.filter(
+        (x) => !new Set(allFonts.map((font) => font.family)).has(x)
+      );
+
+      if (diff && diff.length >= 1) {
+        setShowDialog({ visible: true, missingFont: diff[0] });
+      }
+    };
+
+    checkFonts();
+  }, [storyPages, getFonts]);
```

Comment:
This would run the effect every time the story pages change in the editor. We only want to run this effect once though, so an empty `[]` dependency array is needed (requires suppressing a lint error with an explanation)